proxy/src/console/provider/neon.rs

```diff
@@ -125,6 +130,62 @@ impl Api {
         .await
     }
 
+    async fn do_get_endpoint_jwks(
+        &self,
+        ctx: &RequestMonitoring,
+        endpoint: EndpointId,
+        role: RoleName,
+    ) -> anyhow::Result<Vec<AuthRule>> {
+        if !self
+            .caches
+            .endpoints_cache
+            .is_valid(ctx, &endpoint.normalize())
+            .await
+        {
+            bail!("endpoint not found");
+        }
+        let request_id = ctx.session_id().to_string();
+        async {
+            let request = self
+                .endpoint
+                .get2(|url| {
+                    url.path_segments_mut()
+                        .push("endpoints")
+                        .push(endpoint.as_str())
+                        .push("jwks");
+                })
+                .header("X-Request-ID", &request_id)
```

Comment:
nit: It's possible the compiler will fix the header name during const-eval, but maybe lowercase the "d" or better still, define a HeaderName.

Reply:
hyper will actually fully lowercase the header 😅

Reply:
That's illegal! 😀 Ok, please ignore.